rust/lance/src/index.rs

```diff
@@ -3983,4 +3990,545 @@ mod tests {
+        // Only 5 unique categories
+        let categories =
+            StringArray::from_iter_values((0..num_rows).map(|i| format!("category_{}", i % 5)));
+
+        let struct_array = arrow_array::StructArray::from(vec![
+            (
+                Arc::new(Field::new("status.code", DataType::Int32, false)),
+                Arc::new(status_codes) as Arc<dyn arrow_array::Array>,
+            ),
+            (
+                Arc::new(Field::new("category.name", DataType::Utf8, false)),
+                Arc::new(categories) as Arc<dyn arrow_array::Array>,
+            ),
+        ]);
+
+        let batch =
+            RecordBatch::try_new(schema.clone(), vec![Arc::new(ids), Arc::new(struct_array)])
+                .unwrap();
+
+        let reader = RecordBatchIterator::new(vec![batch].into_iter().map(Ok), schema.clone());
+        let mut dataset = Dataset::write(reader, test_uri, None).await.unwrap();
+
+        // Create Bitmap index on nested field with dots
+        let nested_column_path = "metadata.`status.code`";
+        let params = ScalarIndexParams::default();
+
+        dataset
+            .create_index(
+                &[nested_column_path],
+                IndexType::Bitmap,
+                Some("bitmap_status_idx".to_string()),
+                &params,
+                true,
+            )
+            .await
+            .unwrap();
+
+        // Reload dataset to ensure index is loaded
+        dataset = Dataset::open(test_uri).await.unwrap();
+
+        // Verify index was created
+        let indices = dataset.load_indices().await.unwrap();
+        assert_eq!(indices.len(), 1);
+        assert_eq!(indices[0].name, "bitmap_status_idx");
+
+        // Verify the correct field was indexed
+        let field_id = indices[0].fields[0];
+        let field_path = dataset.schema().field_path(field_id).unwrap();
+        assert_eq!(field_path, "metadata.`status.code`");
+
+        // Test querying with the index and verify it's being used
+        let plan = dataset
+            .scan()
+            .filter("metadata.`status.code` = 5")
+            .unwrap()
+            .explain_plan(false)
+            .await
+            .unwrap();
+
+        // Verify the query plan (scalar indices on nested fields may use optimized filters)
+        // The index may be used internally even if not shown as ScalarIndexQuery
+        assert!(
+            plan.contains("ScalarIndexQuery"),
+            "Query plan should show optimized read. Plan: {}",
+            plan
+        );
+
+        // Also test that the query returns results
+        let results = dataset
+            .scan()
+            .filter("metadata.`status.code` = 5")
+            .unwrap()
+            .try_into_batch()
+            .await
+            .unwrap();
+
+        // Should have ~100 rows with status code 5
+        assert!(results.num_rows() > 0);
+        assert_eq!(results.num_rows(), 100);
```

Comment:
these seem redundant, and the comment seems wrong because the assertion is exact